platform/platform-paxweb-jettyconfig/src/main/java/org/codice/ddf/security/session/AttributeSharingHashSessionIdManager.java

```diff
@@ -54,25 +69,90 @@
 public class AttributeSharingHashSessionIdManager extends HashSessionIdManager {
   // changed this to extend Jetty's impl instead of the abstract class because of a hack in pax web
 
-  private final Map<String, Set<WeakReference<HttpSession>>> sessions = new ConcurrentHashMap<>();
+  private static final Logger LOGGER =
+      LoggerFactory.getLogger(AttributeSharingHashSessionIdManager.class);
+
+  static class SharingSessionInvalidator implements HttpSessionInvalidator {
+
+    private final AttributeSharingHashSessionIdManager idManager;
+
+    SharingSessionInvalidator(AttributeSharingHashSessionIdManager idManager) {
+      this.idManager = idManager;
+    }
+
+    @Override
+    public void invalidateSession(
+        String subjectName, Function<HttpSession, String> sessionSubjectExtractor) {
+
+      final List<AbstractSession> matchingSessions =
+          getHttpSessionStream()
+              .filter(s -> subjectName.equals(sessionSubjectExtractor.apply(s)))
+              .map(AbstractSession.class::cast)
+              .distinct()
+              .collect(Collectors.toList());
+      for (AbstractSession matchingSession : matchingSessions) {
+        idManager.removeSession(matchingSession);
+        if (matchingSession.isValid()) {
+          matchingSession.invalidate();
+        }
+      }
+    }
+
+    private Stream<HttpSession> getHttpSessionStream() {
+      return getFullSessions()
+          .stream()
+          .flatMap(Collection::stream)
+          .map(Reference::get)
+          .filter(Objects::nonNull);
+    }
+
+    private Collection<Set<WeakReference<HttpSession>>> getFullSessions() {
+      return Collections.unmodifiableCollection(idManager.sessionMap.values());
```

Comment:
A little belt and suspenders here. It's not especially expensive to decorate it as unmodifiable so I'm not too worried about it, but it could certainly be removed.